x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+
+        let voter1 = Address::from_str("voter1");
+        let voter2 = Address::from_str("voter2");
+
+        let voters = vec![voter1, voter2];
+
+        test_context.actor = voter1;
+
+        let pid: u32 = simulator
+            .run_step(&Step {
+                endpoint: Endpoint::Execute,
+                method: "propose".to_string(),
+                max_units: u64::MAX,
+                params: vec![
+                    test_context.clone().into(),
+                    Param::FixedBytes(borsh::to_vec(&voters).unwrap()),
+                    program_id.into(),
+                    String::from("pub_last_proposal_id").into(),
+                    Vec::new().into(),
+                    0u64.into(),
+                ],
+            })
+            .unwrap()
+            .result
+            .response()
+            .unwrap();
+
+        let res = simulator
+            .run_step(&Step {
+                endpoint: Endpoint::Execute,
+                method: "execute".to_string(),
+                max_units: u64::MAX,
+                params: vec![test_context.clone().into(), pid.into(), u64::MAX.into()],
+            })
+            .unwrap()
+            .result
+            .response::<Result<(), ProposalError>>()
+            .unwrap();
+
+        assert!(matches!(res, Err(ProposalError::QuorumNotReached)));
+
+        let voter_context = TestContext {
+            actor: voter2,
+            ..test_context.clone()
+        };
+
+        let res = simulator
+            .run_step(&Step {
+                endpoint: Endpoint::Execute,
+                method: "vote".to_string(),
+                max_units: u64::MAX,
+                params: vec![voter_context.clone().into(), pid.into(), true.into()],
+            })
+            .unwrap()
+            .result
+            .response::<Result<(), ProposalError>>()
+            .unwrap();
+
+        assert!(res.is_ok());
+
+        let reached_res = simulator
+            .run_step(&Step {
+                endpoint: Endpoint::ReadOnly,
+                method: "pub_quorum_reached".to_string(),
+                max_units: 0,
+                params: vec![test_context.clone().into(), pid.into()],
+            })
+            .unwrap()
+            .result
+            .response::<Result<bool, ProposalError>>()
+            .unwrap();
+
+        assert!(reached_res.is_ok_and(|reached| reached));
+
+        let res = simulator
+            .run_step(&Step {
+                endpoint: Endpoint::Execute,
+                method: "execute".to_string(),
+                max_units: u64::MAX,
+                params: vec![voter_context.clone().into(), pid.into(), 1000000u64.into()],
+            })
+            .unwrap()
+            .result
+            .response::<Result<DeferDeserialize, ProposalError>>()
+            .unwrap()
+            .unwrap();
+
+        let id_serialized: Vec<u8> = res.deserialize().unwrap();
+        let id: u32 = borsh::from_slice(&id_serialized).unwrap();
```

Comment:
I know that it doesn't look the best but that's because of the `BorshSerialize` that is annotating the `DeferDeserialize`. It should probably also serialize as a white-box.

Reply:
```rust
impl BorshSerialize for DeferDeserialize {
    fn serialize<W: std::io::prelude::Write>(&self, writer: &mut W) -> std::io::Result<()> {
        writer.write_all(&self.0)?;
        Ok(())
    }
}
```

Reply:
Opened https://github.com/ava-labs/hypersdk/issues/1123